### DISCUSSION TITLE
Build production fail Error: resolve-url-loader: CSS error

### DISCUSSION THREAD

**Discussion by hungnguyen1995** [vercel/next.js#11938] [Help] [answered]
Warning: You have enabled experimental feature(s).
Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.

> Using external babel configuration
Creating an optimized production build

Failed to compile.

./static/scss/main.scss
Error: resolve-url-loader: CSS error
  source-map information is not available at url() declaration (found orphan CR, try removeCR option)

> Build error occurred
Error: > Build failed because of webpack errors
    at build (fe-codebase\node_modules\next\dist\build\index.js:13:900)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

____next.config.js

module.exports = withConfig(
    withPlugins(
        [[withBundleAnalyzer], withPwaManifest, withOffline],
        nextConfig
    )
);

How to config in file __ next.config.js ?
removeCR : false

Thanks :(

**Comment by Timer** [comment, marked as answer]
What Next.js version are you on? It may be too old!

We'll need a fully reproducing demo to help you find the bug in your setup.

**Reply by eldieco** [reply]
I just received this same error while trying to deploy within a nginx container on a self-hosted linux server. I'm using next version 9.1.4, and my next.config.js file is a little different than the one above, but I never saw this error before...Did not get this error when building locally or deploying with heroku or now or netlify. Odd.

next.config.js:

```
const withCSS = require('@zeit/next-css');

module.exports = withCSS({
  webpack(config) {
    config.module.rules.push({
      test: /\.(eot|woff|woff2|ttf|svg|png|jpg|gif)$/,
      use: {
        loader: 'url-loader',
        options: {
          limit: 100000,
          name: '[name].[ext]',
        },
      },
    });
    return config;
  },
});

**Reply by eldieco** [reply]
Upgraded to next version 9.3.6, removed next-css, updated next.config.js to not include withCSS and then tried to build on the nginx server and got the following error:

```
(node:16) UnhandledPromiseRejectionWarning: Error: Cannot find module 'webpack'
Require stack:
- /home/eldiedesign/app/node_modules/@next/react-refresh-utils/ReactRefreshWebpackPlugin.js
- /home/eldiedesign/app/node_modules/next/dist/build/webpack-config.js
- /home/eldiedesign/app/node_modules/next/dist/build/index.js
- /home/eldiedesign/app/node_modules/next/dist/cli/next-build.js
- /home/eldiedesign/app/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1014:15)
    at Function.Module._load (internal/modules/cjs/loader.js:884:27)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/home/eldiedesign/app/node_modules/@next/react-refresh-utils/ReactRefreshWebpackPlugin.js:23:17)
    at Module._compile (internal/modules/cjs/loader.js:1185:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1205:10)
    at Module.load (internal/modules/cjs/loader.js:1034:32)
    at Function.Module._load (internal/modules/cjs/loader.js:923:14)
    at Module.require (internal/modules/cjs/loader.js:1074:19)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:16) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:16) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I tested `next build` locally and it built an "optimized production build" successfully.

Not sure why it's not building on my actual server.

**Reply by eldieco** [reply]
OK my problem was I accidentally left a `--production` flag on `npm install`. Once I took that off, the build ran successfully. Whoops. :)

**Comment by stefansdev** [comment]
Had this issue with tailwindcss/jit.
Kept giving me error on prod/build while dev worked fine. Not sure if related to tailwind or what but might help someone. 
I had `mb-[2px]` at the end. Moving it to the middle worked fine.

```
.some_class {
	// ✅ - this works
	@apply grid grid-cols-[auto,1fr,auto] mb-[2px] relative items-center hover:bg-gray-50 hover:shadow;
	// ❌ - this doesnt work
	@apply grid grid-cols-[auto,1fr,auto] relative items-center hover:bg-gray-50 hover:shadow mb-[2px];
}
// NextJS build gives error: "Error: resolve-url-loader: CSS error Invalid mapping: ... "
```

**Reply by tadjohnston** [reply]
Nice find! I'd still be looking for the fix if you hadn't had posted this 👍

**Reply by GorvGoyl** [reply]
@stefansdev have you reported this issue to tailwind?
